modules/programs/vscode.nix

```diff
@@ -48,6 +48,9 @@ let
     // optionalAttrs (!cfg.enableUpdateCheck) { "update.mode" = "none"; }
     // optionalAttrs (!cfg.enableExtensionUpdateCheck) {
       "extensions.autoCheckUpdates" = false;
+    } // optionalAttrs (!cfg.enableExtensionUpdate) {
+      "extensions.autoUpdate" = false;
+      "extensions.autoCheckUpdates" = false;
```

Comment:
It would be nice to emit a warning if settings are conflicting. Evaluation can continue though, with preference for `enableExtensionUpdate` as noted in the description of that option.

Reply:
The issue is that AFAIK there is no way to distinguish the user having explicitly set `enableExtensionUpdateCheck = true;` or having just left the option to the default value, and in the latter case, I don't think we should warn.

Maybe deprecating `enableExtensionUpdateCheck` would be the best way to ensure that there are no conflicting settings in the long run...

Reply:
The warning could state that `enableExtensionUpdate` overrides `enableExtensionUpdateCheck` and that enableExtensionUpdateCheck should be set explicitly to false to avoid the warning.

Reply:
Yes, but if we are going to ask users to set both options to `false`, we could just as well have two separate options, one that corresponds to `extensions.autoUpdate` and one to `extensions.autoUpdate`.